sqlmesh/core/audit/definition.py

```diff
@@ -432,24 +358,109 @@ def text_diff(self, other: Node) -> str:
+            (
+                d.text_diff(meta_a, meta_b, self.dialect),
+                *(
+                    d.text_diff(sa, sb, self.dialect)
+                    for sa, sb in zip_longest(statements_a, statements_b)
+                ),
+                d.text_diff(query_a, query_b, self.dialect),
+            )
+        ).strip()
+
+    def render_definition(self, include_python: bool = True) -> t.List[exp.Expression]:
+        """Returns the original list of sql expressions comprising the model definition.
+
+        Args:
+            include_python: Whether or not to include Python code in the rendered definition.
+        """
+        expressions: t.List[exp.Expression] = []
+        comment = None
+        for field_name in sorted(self.meta_fields):
+            field_value = getattr(self, field_name)
+            field_info = self.all_field_infos()[field_name]
+            if field_name == "standalone" or field_value != field_info.default:
+                if field_name == "description":
+                    comment = field_value
+                else:
+                    expression = exp.Property(
+                        this=field_info.alias or field_name,
+                        value=META_FIELD_CONVERTER.get(field_name, exp.to_identifier)(field_value),
+                    )
+                    if field_name == "name":
+                        expressions.insert(0, expression)
+                    else:
+                        expressions.append(expression)
+
+        audit = d.Audit(expressions=expressions)
+        audit.comments = [comment] if comment else None
+
+        jinja_expressions = []
+        python_expressions = []
+        if include_python:
+            python_env = d.PythonCode(
+                expressions=[
+                    v.payload if v.is_import or v.is_definition else f"{k} = {v.payload}"
+                    for k, v in self.sorted_python_env
+                ]
+            )
+            if python_env.expressions:
+                python_expressions.append(python_env)
+
+            jinja_expressions = self._jinja_macros_to_exprs()
+
+        return [audit, *python_expressions, *jinja_expressions, *self.expressions, self.query]
 
     @property
     def is_audit(self) -> bool:
         """Return True if this is an audit node"""
         return True
 
+    @property
+    def meta_fields(self) -> t.Iterable[str]:
+        return set(AuditMeta.__annotations__) | set(_Node.all_field_infos())
+
     def _create_query_renderer(self, node: _Node) -> QueryRenderer:
-        audit = t.cast(StandaloneAudit, node)
         return QueryRenderer(
             self.query,
             self.dialect,
             self.macro_definitions,
             path=self._path or Path(),
             jinja_macro_registry=self.jinja_macros,
-            python_env=audit.python_env,
+            python_env=self.python_env,
         )
 
+    def _jinja_macros_to_exprs(self) -> t.List[exp.Expression]:
```

Comment:
This looks like mostly a copy paste from `model/definition.py`. Seems like a good candidate from the utility function in `jinja.py` which takes `JinjaMacroRegistry` and returns a list of expressions.

Reply:
I couldn't think of a reason not to add directly to JinjaMacroRegistry, so that's what I did.